src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -221,8 +221,8 @@ private async Task GenerateDeploymentRecipeSnapShot(Recommendation recommendatio
 
             recipe.Id = Guid.NewGuid().ToString();
             recipe.Name = recipeName;
-            recipe.CdkProjectTemplateId = null;
-            recipe.CdkProjectTemplate = null;
+            recipe.CdkProjectTemplateId = string.Empty;
+            recipe.CdkProjectTemplate = string.Empty;
```

Comment:
why do we need empty string? This will have a an affect on JSON serialization and deserialization.

Reply:
They were being explicitly set to null before. In order to preserve the same functionality, I set them to an empty string since I need them to be non nullable. The reason for that is Newtonsoft throws an exception when deserializing the Recipe object when they were set to nullable.

Reply:
We should use `NullValueHandling = NullValueHandling.Ignore` option.

https://www.newtonsoft.com/json/help/html/NullValueHandlingIgnore.htm

Reply:
For some reason, i retested and I didn't run into the error so I reverted the changes related to nullability.